### PR TITLE
infra: Bump version to 0.4.0 

### PR DESCRIPTION
**Pull request by wusyong**
Bump version to 0.4.0 across all crates. And also update some information in README to sync with other different language libraries. There are some fields in README are not complete yet. But I would like to publish version soon to make sure all this process works fine. I will fully resolve it in next PR.

### CODE REVIEW

**Comment by njaremko** [comment]
Why 0.2 -> 0.4? 0.3 seems to be unused.

Also, saying this official and such, you're implying that people should use the library with actual Iota, but this library hasn't been audited and it's test suite isn't very large, so I think that's irresponsible.

**Comment by wusyong** [comment]
Right, I do mention it's beta software. But since we are transiting to bee as dependencies, it's better to back with WIP title in this case. I changed to this format to match same template as other language library.

File: Cargo.toml [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+authors = ["Nathan Jaremko <nathan@jaremko.ca>", "Yu-Wei Wu <wusyong9104@gmail.com>"]
 license = "MIT"
 readme = "README.md"
 keywords = ["iota", "api"]
-repository = "https://github.com/njaremko/iota-lib-rs"
-homepage = "https://github.com/njaremko/iota-lib-rs"
+repository = "https://github.com/iotaledger/iota.rs"
+homepage = "https://github.com/iotaledger/iota.rs"
 documentation = "https://docs.rs/iota-lib-rs"
 description = "A rust implementation of the IOTA API"
```
Comment (oopsmonk) [inline Cargo.toml]:
```suggestion
description = "A rust implementation of the IOTA Client API"
```

File: README.md [hunk 2]
```diff
@@ -1,82 +1,68 @@
...
+[![License](https://img.shields.io/badge/license-MIT-blue.svg)](https://github.com/iotaledger/iota-lib-rs/blob/master/LICENSE)
+
+This is the **official** Rust client library, which allows you to do the following:
+* Create transactions
+* Sign transactions
+* Generate addresses
+* Interact with an IRI node
+
+This is beta software, so there may be performance and stability issues.
+Please report any issues in our [issue tracker](https://github.com/iotaledger/iota-lib-rs/issues).
```
Comment (oopsmonk) [inline README.md]:
```suggestion
Please report any issues in our [issue tracker](https://github.com/iotaledger/iota.rs/issues).
```

File: README.md [hunk 3]
```diff
@@ -1,82 +1,68 @@
...
-        - [x] get_inputs
-        - [x] prepare_transfers
-        - [x] traverse_bundle
-        - [x] send_transfer
-        - [x] get_bundle
-
-Here's an example of how to send a transaction: (Note that we're using the address as the seed in `send_transfer()`...don't do this)
+[![License](https://img.shields.io/badge/license-MIT-blue.svg)](https://github.com/iotaledger/iota-lib-rs/blob/master/LICENSE)
+
+This is the **official** Rust client library, which allows you to do the following:
```
Comment (oopsmonk) [inline README.md]: not yet. 
```suggestion
This is a Rust client library, which allows you to do the following:
```

File: README.md [hunk 4]
```diff
@@ -1,82 +1,68 @@
...
+* Create transactions
+* Sign transactions
+* Generate addresses
+* Interact with an IRI node
+
+This is beta software, so there may be performance and stability issues.
```
Comment (oopsmonk) [inline README.md]:
```suggestion
This client library is still in the beta stage, so there may be performance and stability issues.
```

File: README.md [hunk 5]
```diff
@@ -1,82 +1,68 @@
...
+iota-lib-rs = "0.4"
+```
+
+## Getting started
+
+After you've [installed the library](#installing-the-library),  you can connect to an IRI node to send transactions to it and interact with the ledger.
+
+To connect to a local IRI node, we provide a module `Client` :
+
+```js
```
Comment (oopsmonk) [inline README.md]:
```suggestion
```rust
```

**Review by oopsmonk — CHANGES_REQUESTED** [review]
@wusyong you published this change to create.io but it should be after this PR merged.

File: README.md [hunk 6]
```diff
@@ -1,82 +1,68 @@
...
-Here's an example of how to send a transaction: (Note that we're using the address as the seed in `send_transfer()`...don't do this)
+[![License](https://img.shields.io/badge/license-MIT-blue.svg)](https://github.com/iotaledger/iota-lib-rs/blob/master/LICENSE)
+
+This is the **WIP** Rust client library, which allows you to do the following:
+* Create transactions
+* Sign transactions
+* Generate addresses
+* Interact with an IRI node
+
+This is client library is still in the beta stage, so there may be performance and stability issues. As IOTA Foundation currently working on `bee`, we also decided to re-implement common libraries for security. This library is going to be feature freeze untill fundamental crates are done.
```
Comment (oopsmonk) [inline README.md]:
```suggestion
This client library is still in the beta stage, so there may be performance and stability issues. As IOTA Foundation currently working on `bee`, we also decided to re-implement common libraries for security. This library is going to be feature freeze untill fundamental crates are done.
```

File: README.md [hunk 7]
```diff
@@ -110,7 +96,12 @@ fn main() {
 }
 ```
 
-# Donations:
-If you feel so inclined, you can find my address for donations at:
+## Supporting the project
+
+## Joining the discussion
+
+If you want to get involved in the community, need help with getting setup, have any issues related with the library or just want to discuss IOTA, Distributed Registry Technology (DRT) and IoT with other people, feel free to join our [Discord](https://discord.iota.org/).
```
Comment (oopsmonk) [inline README.md]:
```suggestion
If you want to get involved in the community, need help with setting up, have any issues related to the library or just want to discuss IOTA, Distributed Ledger Technology (DLT) and IoT with other people, feel free to join our [Discord](https://discord.iota.org/).
```

**Review by oopsmonk — CHANGES_REQUESTED** [review]

**Review by oopsmonk — APPROVED** [review]

**Pull request merged**